### PR TITLE
[FLINK-21343] Update documentation regarding unified savepoint format

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change
Updates the documentation about the possibility to migrate from one state backend to another.

Should be merged after: #14913 and #14925

## Verifying this change

The documentation can be compiled.

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: docs/content/docs/ops/state/savepoints.md [hunk 1]
```diff
@@ -46,9 +46,9 @@ In contrast to all this, Savepoints are created, owned, and deleted by the user.
...
 
-Those conceptual differences aside, the current implementations of Checkpoints and Savepoints are basically using the same code and produce the same format. However, there is currently one exception from this, and we might
-introduce more differences in the future. The exception is incremental checkpoints with the RocksDB state backend. They are using some RocksDB internal format instead of Flink’s native savepoint format. This makes them the
-first instance of a more lightweight checkpointing mechanism, compared to Savepoints.
+In Flink 1.13 we unified the binary format of savepoints across all state backends. That means you can take a savepoint and then restore from it using a different state backend.
+All the state backends produce a common format only starting from the version 1.13. You cannot do that with checkpoints. It means the savepoint binary format might be different from the one used for checkpoints. This is the case e.g. for Heap state backend or
+if you use incremental checkpoints with the RocksDB state backend. The RocksDB full snapshot share the format with unified savepoints, but it is not guaranteed, and we might introduce further modifications in the future.
```
Comment (sjwiesman) [inline docs/content/docs/ops/state/savepoints.md:51]:
```suggestion
Flink's savepoint binary format is unified across all state backends. That means you can take a savepoint with one state backend and then restore it using another.

{{< hint info >}}
State backends did not start producing a common format until version 1.13. Therefore, if you want to switch the state backend you should first upgrade your Flink version than
take a savepoint with the new version, and only after that, you can restore it with a different state backend.
{{< /hint >}}
```

File: docs/content/docs/ops/state/state_backends.md [hunk 2]
```diff
@@ -136,15 +136,16 @@ The total memory amount of RocksDB instance(s) per slot can also be bounded, ple
...
 When deciding between `FsStateBackend` and `RocksDB`, it is a choice between performance and scalability.
 `FsStateBackend` is very fast as each state access and update operates on objects on the Java heap; however, state size is limited by available memory within the cluster.
 On the other hand, `RocksDB` can scale based on available disk space and is the only state backend to support incremental snapshots.
 However, each state access and update requires (de-)serialization and potentially reading from disk which leads to average performance that is an order of magnitude slower than the memory state backends.
 
+In Flink 1.13 we unified the binary format of Flink's savepoints. That means you can take a savepoint and then restore from it using a different state backend.
+All the state backends produce a common format only starting from the version 1.13. Therefore, if you want to switch the state backend you should first upgrade your Flink version than
+take a savepoint with the new version and only after that you can restore it with a different state backend.
+
```
Comment (sjwiesman) [inline docs/content/docs/ops/state/state_backends.md:148]:
```suggestion
{{< hint info >}}
In Flink 1.13 we unified the binary format of Flink's savepoints. That means you can take a savepoint and then restore from it using a different state backend.
All the state backends produce a common format only starting from version 1.13. Therefore, if you want to switch the state backend you should first upgrade your Flink version than
take a savepoint with the new version, and only after that you can restore it with a different state backend.
{{< /hint >}}
```

**Review by sjwiesman — COMMENTED** [review]
More of a nit, but I like to avoid `since version` statements as we can assume users are reading the version of the docs for the version they are using. If instead, the upgrade strategy goes into a hint it stands out for those users who need it without cluttering the main content.

**Comment by dawidwys** [comment]
Thanks for the review @sjwiesman !

**Pull request closed without merging**